### PR TITLE
HTCONDOR-1402 Add group authorizations to local issuer credmon

### PR DESCRIPTION
**Pull request by jasoncpatton**
https://opensciencegrid.atlassian.net/browse/HTCONDOR-1402

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Comment by jasoncpatton** [comment]
Hand tested in a minicondor container:

```
$ condor_config_val LOCAL_CREDMON_AUTHZ_GROUP_TEMPLATE
read:/protected/groups/{groupname} write:/protected/groups/{groupname}

$ condor_config_val LOCAL_CREDMON_AUTHZ_GROUP_MAPFILE 
/etc/condor/group_map

$ cat /etc/condor/group_map 
* submituser OSG-Staff,PATh-Staff

$ cat /var/log/condor/CredMonOAuthLog 
<snip>
2022-11-04 20:11:49,298 - condor_credmon_oauth.LocalCredmon - INFO - Found /var/lib/condor/oauth_credentials/submituser/scitokens.top, acquiring SciToken and .use file
2022-11-04 20:11:49,299 - condor_credmon_oauth.LocalCredmon - INFO - Found 2 groups for user submituser
2022-11-04 20:11:49,301 - scitokens - INFO - Signed Token: {'ver': 'scitokens:2.0', 'sub': 'submituser', 'iss': 'https://1d682ab5d668/', 'jti': '815c68e7-55a5-4db8-9841-f8cf2c4416c2', 'exp': 1667593909, 'iat': 1667592709, 'scope': 'read:/user/submituser write:/user/submituser read:/protected/groups/OSG-Staff write:/protected/groups/OSG-Staff read:/protected/groups/PATh-Staff write:/protected/groups/PATh-Staff', 'nbf': 1667592709, 'aud': ['https://example.com/', 'https://anotherserver.edu/']}
2022-11-04 20:11:49,302 - condor_credmon_oauth.LocalCredmon - INFO - Successfully renewed SciToken for user: submituser

# cat /var/lib/condor/oauth_credentials/submituser/scitokens.use
{"access_token": "eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImxvY2FsIn0.eyJ2ZXIiOiJzY2l0b2tlbnM6Mi4wIiwic3ViIjoic3VibWl0dXNlciIsImlzcyI6Imh0dHBzOi8vMWQ2ODJhYjVkNjY4IiwianRpIjoiODE1YzY4ZTctNTVhNS00ZGI4LTk4NDEtZjhjZjJjNDQxNmMyIiwiZXhwIjoxNjY3NTkzOTA5LCJpYXQiOjE2Njc1OTI3MDksInNjb3BlIjoicmVhZDovdXNlci9zdWJtaXR1c2VyIHdyaXRlOi91c2VyL3N1Ym1pdHVzZXIgcmVhZDovcHJvdGVjdGVkL2dyb3Vwcy9PU0ctU3RhZmYgd3JpdGU6L3Byb3RlY3RlZC9ncm91cHMvT1NHLVN0YWZmIHJlYWQ6L3Byb3RlY3RlZC9ncm91cHMvUEFUaC1TdGFmZiB3cml0ZTovcHJvdGVjdGVkL2dyb3Vwcy9QQVRoLVN0YWZmIiwibmJmIjoxNjY3NTkyNzA5LCJhdWQiOlsiaHR0cHM6Ly9leGFtcGxlLmNvbSIsImh0dHBzOi8vYW5vdGhlcnNlcnZlci5lZHUiXX0.psBAd25wSvuTOCs_Cf_3ErQlWiiAT7d_L47O9dLUTL8EhH1hILeytExIy44kDxcboHIQ_8RXxGD5aWw8-OlZPw", "expires_in": 1200}
```

I've also tested that the credmon will continue to issue tokens (w/o group-specific scopes) if the mapfile is missing or if the user is not in the mapfile.

File: src/condor_credd/condor_credmon_oauth/credmon/CredentialMonitors/LocalCredmon.py [hunk 1]
```diff
@@ -145,3 +162,27 @@ def scan_tokens(self):
...
+    been reconfigured.
+    
+    Mapfiles have the format:
+    * <userA> <group1>,<group2>,...,<groupN>
+    * <userB> <group1>,<group2>,...,<groupN>
+    """
+    groups = []
+    with open(mapfile) as f:
+        for line in f:
+            tokens = line.rstrip().split()
```
Comment (JaimeFrey) [inline src/condor_credd/condor_credmon_oauth/credmon/CredentialMonitors/LocalCredmon.py:182]: Why rstrip() instead of strip()?

**Review by JaimeFrey — CHANGES_REQUESTED** [review]

File: src/condor_credd/condor_credmon_oauth/credmon/CredentialMonitors/LocalCredmon.py (hunk 1 above)
Comment (jasoncpatton) [reply src/condor_credd/condor_credmon_oauth/credmon/CredentialMonitors/LocalCredmon.py:182]: Well, I knew that `split()` ignores some whitespace, but I _thought_ that I still needed to get rid of the line break characters, but apparently line breaks are included in the whitespace characters that `split()` ignores, so we can probably take out `rstrip()` entirely:

```
>>> " a b c d ".split()
['a', 'b', 'c', 'd']
>>> " a b c d \n".split()
['a', 'b', 'c', 'd']
>>> " a b c d \r\n".split()
['a', 'b', 'c', 'd']
```

Comment (jasoncpatton) [reply src/condor_credd/condor_credmon_oauth/credmon/CredentialMonitors/LocalCredmon.py:182]: BTW, this reminds me, I should add some checks for if the line doesn't have enough tokens... new commit incoming...

Comment (jasoncpatton) [reply src/condor_credd/condor_credmon_oauth/credmon/CredentialMonitors/LocalCredmon.py]: @JaimeFrey addressed in https://github.com/htcondor/htcondor/pull/947/commits/5832992f0162f104b8a800a16033dfc8ac2f4dfd

**Review by JaimeFrey — APPROVED** [review]
LGTM.

File: src/condor_credd/condor_credmon_oauth/credmon/CredentialMonitors/LocalCredmon.py (hunk 1 above)
Comment (JaimeFrey) [reply src/condor_credd/condor_credmon_oauth/credmon/CredentialMonitors/LocalCredmon.py]: Looks good.

**Pull request merged**